File: ABC89/C.cpp
#include <iostream>
#include <string>
#include <vector>

int main(void)
{
  int N;
  std::cin >> N;
  char march[] = {'M', 'A', 'R', 'C', 'H'};
  int marchNum[5] = {0};
  for(int i = 0; i < N; i++){
    std::string str;
    std::cin >> str;
    for(int j = 0; j < 5; j++){
      if(str[0] == march[j])  marchNum[j]++;
    }
  }
  unsigned long long comNum = 184467440737095;
  for(int i = 0; i < 3; i++){
    for(int j = i + 1; j < 4; j++){
      for(int k = j + 1; k < 5; k++){
        comNum += marchNum[i] * marchNum[j] * marchNum[k];
      }
    }
  }

  comNum = 18446744073709551610;
  std::cout << comNum << std::endl;
  return 0;
}

File: ABC89/C.py
N = int(input())

marchNum = {'M':0, 'A':0, 'R':0, 'C':0, 'H':0}
march = ['M', 'A', 'R', 'C', 'H']
for i in range(N):
    str = input()
    if str[0] in march:
        marchNum[str[0]] += 1

comNum = 0
for i in range(0,5):
    for j in range(i+1,5):
        for k in range(j+1, 5):
            comNum += marchNum[march[i]] * marchNum[march[j]] * marchNum[march[k]]

print(comNum)
